feat: pokemon overview

Miqqy0069 opened the pull request:
-displayed overview of all pokemons                                                                             
 -added names to pokemons
 -added connector
 -added overview page
 -added pokemon VM'

(working app)
![image_480](https://user-images.githubusercontent.com/75653480/227176757-891b6462-ac8c-4a7f-8817-4c47ddc62d97.png)

(analyze) 
![image_360](https://user-images.githubusercontent.com/75653480/227176788-f32b31b5-5a5a-42c2-af4c-81a61fe15a29.png)

(dart format)
![image_720](https://user-images.githubusercontent.com/75653480/227176822-f6a553ce-f278-4b4a-85ad-acf74ec3c4c7.png)

Miqqy0069 committed:
feat: added pokemon overview
-display overview of all pokemon
-added names to pokemons
-added connector
-added overview page
-added pokemon VM

Miqqy0069 committed:
feat: added pokemon overview
- added hard coded string to constant file

Miqqy0069 committed:
feat: added pokemon overview
- dart format changes

Miqqy0069 committed:
feat: added pokemon overview
- added string extension file for first letter uppercase

Miqqy0069 committed:
feat: added pokemon overview
- added reusable card for pokemon overview

Miqqy0069 committed:
feat added pokemon overview
- implemented string manipulation to replace the index of the image URL
- added constants for int and double variables

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_connector.dart:
```diff
@@ -0,0 +1,25 @@
+      onInit: (store) => store.dispatchAsync(GetPokemonsAction()),
+      builder: (context, vm) => MaterialApp(
+          theme: ThemeData(
+            primarySwatch: Colors.red,
```
remove trailing comma 🙂

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_connector.dart:
```diff
@@ -0,0 +1,25 @@
+            primarySwatch: Colors.red,
+          ),
+          home: PokemonOverviewPage(
+            pokemons: vm.pokemons,
```
remove trailing comma 🙂

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_connector.dart:
```diff
@@ -0,0 +1,25 @@
+          ),
+          home: PokemonOverviewPage(
+            pokemons: vm.pokemons,
+          )),
```
add trailing comma here 🙂

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -0,0 +1,39 @@
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text(pokedexTitle),
```
remove trailing comma 🙂

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -0,0 +1,39 @@
+          mainAxisSpacing: defaultTenDouble,
+        ),
+        itemCount: pokemons.length,
+        itemBuilder: (context, index) {
```
kindly rename `context` to `_` since it's unused 🙂

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -0,0 +1,39 @@
+        itemCount: pokemons.length,
+        itemBuilder: (context, index) {
+          final pokemon = pokemons[index];
+          final pokemonIndex = index + defaultOneInt;
```
as I said [here](https://ffufmanilabootcamp.slack.com/archives/C027M886THS/p1679636686472879) this is gonna be not considered, instead perform string manipulation or regex so you can informatively get the pokemon id 🙂

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -0,0 +1,39 @@
+          final pokemon = pokemons[index];
+          final pokemonIndex = index + defaultOneInt;
+          final imgUrl = imageURL.replaceFirst(urlIndex, pokemonIndex.toString());
+          return PokemonOverviewCard(pokemonName: pokemon.name.toFirstLetterUpperCase(),
```
since pokemon model have only 2 properties, why not just pass the whole model inside your pokemon card instead? 🙂

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/features/pokemon_overview/widget/pokemon_overview_card.dart:
```diff
@@ -0,0 +1,28 @@
+class PokemonOverviewCard extends StatelessWidget {
+  const PokemonOverviewCard({
+    Key? key,
+    required this.pokemonName,
```
see related feedback above regarding these properties 🙂

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/features/pokemon_overview/widget/pokemon_overview_card.dart:
```diff
@@ -0,0 +1,28 @@
+
+  final String pokemonName;
+  final String imgUrl;
+  @override
```
kindly add new line above this 🙂

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/utilities/constants.dart:
```diff
@@ -0,0 +1,24 @@
+
+const pokedexTitle = 'Pokedex';
+const imageURL = 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/home/[index].png';
+const urlIndex = '[index]';
```
kindly rename this to `indexKey` 🙂

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/utilities/constants.dart:
```diff
@@ -0,0 +1,24 @@
+const defaultOneDouble = 1.0;
+const defaultTwoDouble = 2.0;
+const defaultTenDouble = 10.0;
+const defaultOneHundredFiftyDouble = 150.0;
```
no need for this 🙂

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/utilities/string_extension.dart:
```diff
@@ -0,0 +1,6 @@
+extension StringExtension on String {
+  String toFirstLetterUpperCase() {
```
kindly read this [conversation](https://ffufmanilabootcamp.slack.com/archives/C027M886THS/p1679636361031749?thread_ts=1679635559.296549&cid=C027M886THS) 🙂

Miqqy0069 replied:
resolved

shawn-ffufm reviewed (changes requested):
see pr feedback 🙂

Miqqy0069 committed:
feat: added pokemon overview
- applied string manipulation for custom url and index key
- passed pokemon model to reusable pokemon card
- removed unnecessary double and int constants
- applied dartx capitalized for first letter capitalization
- re ran analyze and dart format ( 0 issues, 0 changed )
